merkledag/node.go

```diff
@@ -107,9 +110,17 @@ func (n *Node) AddNodeLinkClean(name string, that *Node) error {
 	if err != nil {
 		return err
 	}
-	lnk.Name = name
+	n.AddRawLink(name, lnk)
 
+	return nil
+}
+
+// AddRawLink adds a pre-provided link to this node
+func (n *Node) AddRawLink(name string, lnk *Link) error {
```

Comment:
good catch, i'd be more comfortable making a copy of the link before assigning it... just to be safe.